rdflib/plugins/stores/sparqlconnector.py

```diff
@@ -94,47 +92,44 @@ def query(self, query, default_graph=None):
 
         if self.method == "GET":
             args["params"].update(params)
+            qsa = "?" + urlencode(args["params"])
+            res = urlopen(Request(self.query_endpoint + qsa, headers=args["headers"]))
         elif self.method == "POST":
             args["headers"].update({"Content-Type": "application/sparql-query"})
-            args["data"] = params
+            try:
+                res = urlopen(Request(self.query_endpoint, data=query.encode(), headers=args["headers"]))
```

Comment:
Don't you need `method='POST'` here?

Reply:
No, if you specify `data=...` then urllib.Request becomes a POST automatically

Reply:
https://docs.python.org/3/library/urllib.request.html#urllib.request.Request